.github/workflows/release.yml

```diff
@@ -0,0 +1,34 @@
+name: envattrs release workflow
+
+on:
+  push:
+    tags:
+      - '*'
```

Comment:
shouldn't this be on `release/published` rather than `tags`?

Reply:
maybe it's a decision that this project needs to make and write down.
making a release creates a tag (unless you select one manually), so it's often equivalent.
at the same time, what if someone pushes a tag on a branch? (a release cab be made from a branch too).

I've seen both variants -- push artefact on tag, so that library users can depend on a specific tag; and publish official release on release only, as there's a separate permission bit for releases.
Either will do as long as it's agreed upon.

What bugs me is that the tag/release name is in not connected to sw version at PYPI.

Reply:
> What bugs me is that the tag/release name is in not connected to sw version at PYPI.

we could presumably add a github action check for this?

Reply:
The way I set it up in my projects:
* on tag (or on release)
* restricted to the main branch (a bit tricky)
* publish with the tag name

It's automated, neat, no confusion.

OTOH this project may see it fit that versions are advanced manually, in which case, a check may be more appropriate.

Reply:
Or maybe automate something like https://pypi.org/project/poetry-publish/ ? 🤔

Reply:
I've updated it as so:
- Run on released
- Restricted to default branch (master)
- Release tag name must match package version